src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -833,6 +848,46 @@ public static ComponentResultEntity toEntity(LoadFlowResult.ComponentResult comp
                 .bodyToMono(String.class);
     }
 
+    public Mono<String> getSecurityAnalysisStatus(String studyName, String userId) {
+        Objects.requireNonNull(studyName);
+        Objects.requireNonNull(userId);
+
+        return studyRepository.findStudy(userId, studyName).flatMap(entity -> {
+            UUID resultUuid = entity.getSecurityAnalysisResultUuid();
+            return Mono.justOrEmpty(resultUuid).flatMap(uuid -> {
+                String path = UriComponentsBuilder.fromPath(DELIMITER + SECURITY_ANALYSIS_API_VERSION + "/results/{resultUuid}/status")
+                        .buildAndExpand(resultUuid)
+                        .toUriString();
+                return webClient
+                        .get()
+                        .uri(securityAnalysisServerBaseUri + path)
+                        .retrieve()
+                        .onStatus(httpStatus -> httpStatus == HttpStatus.NOT_FOUND, clientResponse -> Mono.error(new StudyException(SECURITY_ANALYSIS_NOT_FOUND)))
+                        .bodyToMono(String.class);
+            });
+        });
+    }
+
+    public Mono<Void> setSecurityAnalysisStatus(String studyName, String userId, SecurityAnalysisStatus status) {
```

Comment:
To remove, we don't want to set the result status from API, it is supposed to be a computation state.

Reply:
Changed to 'invalidateSecurityAnalysisStatus'